src/game_objects/tile.cpp

```diff
@@ -0,0 +1,70 @@
+#include "tile.h"
+#include "util/colors.h"
+
+Tile::Tile() :
+    construct(nullptr) {}
+
+Tile::~Tile() {}
+
+// TODO: GET RID OF THIS. DO NOT OVERRIDE GAMEOBJECT::DRAW.
+void Tile::draw(Shader *shader, SceneInfo &scene_info) {
+    GameObject::draw(shader, scene_info);
+    if (construct) construct->draw(shader, scene_info);
+}
+
+FloorTile::~FloorTile() {}
+
+WallTile::~WallTile() {}
+
+FloorTile::FloorTile(int x, int z) : Tile::Tile() {
+    width = 1;
+    height = 1;
+    this->x = x;
+    this->z = z;
+    buildable = true;
+
+    Mesh* mesh = new Mesh();
+
+    mesh->local_transform = glm::mat4(1.0f);
+    mesh->geometry = GeometryGenerator::generate_plane(0.5f, 0);
+    Material *mat = new Material(color::indian_red);
+    mesh->material = mat;
+
+    model->add_mesh(mesh);
+
+    transform.set_position((float)x, 0.0f, (float)z);
+}
+
+void FloorTile::update() {
+}
+
+WallTile::WallTile(int x, int z) : Tile::Tile() {
+    width = 1;
+    height = 1;
+    this->x = x;
+    this->z = z;
+    buildable = false;
+
+    Mesh* mesh = new Mesh();
+
+    mesh->local_transform = glm::mat4(1.0f);
```

Comment:
Again, comment? Also maybe leave a note on why that line below is commented out?